File: CP/Java/A_Fair_Playoff.java
//https://codeforces.com/problemset/problem/1535/A

import java.util.Scanner;
public class A_Fair_Playoff {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int t = scanner.nextInt(); // Number of test cases

        for (int i = 0; i < t; i++) {
            int x = scanner.nextInt();
            int z = scanner.nextInt();
            int y = scanner.nextInt();
            int e = scanner.nextInt();

            if ((Math.max(x, z) < Math.min(y, e)) || (Math.max(y, e) < Math.min(x, z))) {
                System.out.println("NO");
            } else {
                System.out.println("YES");
            }
        }
    }
}

/*
Input
4
3 7 9 5
4 5 6 9
5 3 8 1
6 5 3 2

Output
YES
NO
YES
NO

 */

File: CP/CPP/A_Fair_Playoff.cpp
//https://codeforces.com/problemset/problem/1535/A

#include<iostream>
using namespace std;
int main()
{
    int n,match1Winner, match2Winner, match1Loser,match2Loser;
    cin>>n;

    int arr[n][4];

    string ans[n];

    for(int i = 0; i<n; i++)
    {
        for(int j = 0; j< 4; j++)
        {
            cin>>arr[i][j];
        }
    }

    for(int i = 0; i<n; i++)
    {
        match1Winner = max(arr[i][0],arr[i][1]);
        match2Winner = max(arr[i][2], arr[i][3]);

        match1Loser = min(arr[i][0],arr[i][1]);
        match2Loser = min(arr[i][2], arr[i][3]);

        if(match1Winner<match2Loser || match2Winner<match1Loser)
            ans[i] = "NO";
        else
            ans[i] = "YES";
    }

    for(int i = 0; i<n; i++)
    {
        cout<<ans[i]<<endl;
    }
return 0;
}

/*

input
4
3 7 9 5
4 5 6 9
5 3 8 1
6 5 3 2

output
YES
NO
YES
NO

*/